pkg/addons/data/ClusterRoleBinding.rbac.authorization.k8s.io/admin-reconciler.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: admin-reconciler
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: admin
```

Comment:
Isn't this cluster admin acess to the service account? Intentional?

Reply:
I am asking whether it makes sense to grant cluster admin access to the dedicated-admin-reconciler.

Reply:
I agree that this is strange but this is granted at the project level and in order for the reconciler to grant access for dedicated-admins he also must be an admin.

Reply:
It can probably be a RoleBinding instead of a ClusterRoleBinding?

Reply:
This admin refers to the clusterrole admin not to the clusterrole cluster-admin.  When a rolebinding is created in a project to a clusterrole, the rolebinding only applies to that project.

Let's summarize this statement:
We are granting a serviceaccount in the openshift-infra namespace named dedicated-admin-reconciler the clusterrole admin through this dedicated-admin-reconciler clusterrolebinding.  This allows for the system:serviceaccounts:openshift-infra:dedicated-admin-reconciler to perform actions at the cluster level, e.g. when new projects are created.  If this was a rolebinding instead of a clusterrolebinding then the service account would not be able to grant permissions on newly created projects.

This explanation might be more for me than anyone else.

I verified this with a test:
```
+ oc auth reconcile -f /tmp/rolebindings -n test4
Project: test4
rolebinding.rbac.authorization.k8s.io "dedicated-project-admin" reconciled
Error from server (Forbidden): rolebindings.rbac.authorization.k8s.io "admin" is forbidden: attempt to grant extra privileges:
Error from server (Forbidden): rolebindings.rbac.authorization.k8s.io "view" is forbidden: attempt to grant extra privileges:
```

Reply:
Yes, this is fine, I confused admin in a clusterrolebinding with a cluster admin.